src/sdk/namespace_fs.js

```diff
@@ -543,8 +608,16 @@ class NamespaceFS {
         // Uploading to a temp file then we will rename it. 
         const upload_id = uuidv4();
         const upload_path = path.join(this.bucket_path, this.get_bucket_tmpdir(), 'uploads', upload_id);
+        const xattr = _.isEmpty(params.xattr) ? undefined : params.xattr;
```

Comment:
If we are making fs_xattr then lets call it like that, and also move the empty check into the function to make it easier to prepare:
```suggestion
        const fs_xattr = get_fs_xattr_for_write(params.xattr);
```